Source/WebCore/rendering/RenderBox.h

```diff
@@ -739,9 +740,9 @@ override;
     enum class MinimumSizeIsAutomaticContentBased { Yes, No };
     void constrainLogicalMinMaxSizesByAspectRatio(LayoutUnit& minSize, LayoutUnit& maxSize, LayoutUnit computedSize, MinimumSizeIsAutomaticContentBased, ConstrainDimension) const;
 
-    static LayoutUnit blockSizeFromAspectRatio(LayoutUnit borderPaddingInlineSum, LayoutUnit borderPaddingBlockSum, double aspectRatio, BoxSizing boxSizing, LayoutUnit inlineSize)
+    static LayoutUnit blockSizeFromAspectRatio(LayoutUnit borderPaddingInlineSum, LayoutUnit borderPaddingBlockSum, double aspectRatio, BoxSizing boxSizing, LayoutUnit inlineSize, AspectRatioType aspectRatioType, bool isReplacedElementWithIntrinsicRatio)
     {
-        if (boxSizing == BoxSizing::BorderBox)
+        if (boxSizing == BoxSizing::BorderBox && aspectRatioType == AspectRatioType::Ratio && !isReplacedElementWithIntrinsicRatio)
```

Comment:
I believe it should be ok here to leave it out. If it auto && ratio, then this check will fail and it will use the ContentBox dimensions. 

However, now that I'm looking at the spec again, maybe this should just be `boxSizing == BoxSizing::BorderBox && aspectRatioType == AspectRatioType::Ratio`? Since the only time we need the BorderBox size is when aspect-ratio is `<ratio>` and the box-sizing property is set to border-box.

Thoughts?

https://w3c.github.io/csswg-drafts/css-sizing-4/#aspect-ratio

Reply:
Yes, I agree with you! Should be `boxSizing == BoxSizing::BorderBox && aspectRatioType == AspectRatioType::Ratio`.

Reply:
So it looks like the reason I had this additional check is because css-sizing/aspect-ratio/replaced-element-014.html seems to assert that replaced elements should *always* use the content box sizing. Maybe we can change this argument to take a bool and pass in `isRenderReplaced()`?

I think that would allow us to get rid of `isReplacedElementWithIntrinsicRatio` and only add in the necessary logic in the couple places where it is needed.

Reply:
I couldn't completely get rid of the method, but I renamed is to `isRenderReplacedWithAspectRatio`. The method is not used as much as it was before, but it still needs to actually compute the intrinsic ratio. I tried to add the intrinsic ratio as a member field and update it, but it wasn't completely straightforward. It will take a little bit of work to determine when we should actually update the intrinsic ratio and then update it accordingly.

I think that work should probably be done outside the scope of this patch and hopefully we could land this without that logic. If you disagree, though, let me know and I can perhaps see what can be done!